### PR TITLE
Upstream and replace `github-actions-runner` with `eks/actions-runner-controller`

### PR DESCRIPTION
**Pull request by milldr**
## what
- `github-actions-runner` is outdated and deprecated
- `eks/actions-runner-controller` is how we deploy GitHub runners to EKS

## why
- Deploying GitHub Runners to EKS

### CODE REVIEW

File: modules/eks/actions-runner-controller/README.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+| Name | Description |
+|------|-------------|
+| <a name="output_metadata"></a> [metadata](#output\_metadata) | Block status of the deployed release |
+| <a name="output_metadata_action_runner_releases"></a> [metadata\_action\_runner\_releases](#output\_metadata\_action\_runner\_releases) | Block statuses of the deployed actions-runner chart releases |
+| <a name="output_webhook_payload_url"></a> [webhook\_payload\_url](#output\_webhook\_payload\_url) | Payload URL for GitHub webhook |
+<!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+
+## References
+
```
Comment (milldr) [inline modules/eks/actions-runner-controller/README.md:187]: add reference to upstream module

File: modules/eks/actions-runner-controller/charts/actions-runner/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "1.16.0"
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/charts/actions-runner/Chart.yaml:24]: Please update the app version
```suggestion
# This chart only deploys Resources for actions-runner-controller, so app version does not really apply.
# We use Resource API version instead.
appVersion: "v1alpha1"
```

File: modules/eks/actions-runner-controller/charts/actions-runner/templates/runnerdeployment.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      {{- range .Values.labels }}
+        - {{ . | quote }}
+      {{- end }}
+      # dockerdWithinRunnerContainer = false means access to a Docker daemon is provided by a sidecar container.
+      dockerdWithinRunnerContainer: {{ .Values.dind_enabled }}
+      image: {{ .Values.image | quote }}
+      imagePullPolicy: IfNotPresent
+      serviceAccountName: {{ .Values.service_account_name }}
+      resources:
+        {{- toYaml .Values.resources | nindent 8 }}
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/charts/actions-runner/templates/runnerdeployment.yaml:46]:
```suggestion
        {{- toYaml .Values.resources | nindent 8 }}
 
```

File: modules/eks/actions-runner-controller/variables.tf [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+  kebab-case.
+
+  For example:
+
+  ```hcl
+  organization_runner = {
+    type = "organization" # can be either 'organization' or 'repository'
+    dind_enabled: false # A Docker sidecar container will be deployed
+    image: summerwind/actions-runner # If dind_enabled=true, set this to 'summerwind/actions-runner-dind'
+    scope = "ACME"
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/variables.tf:131]:
```suggestion
    scope = "ACME" # org name for Organization runners, repo name for Repository runners
```

File: modules/eks/actions-runner-controller/README.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+```yaml
+components:
+  terraform:
+    eks/actions-runner-controller:
+      settings:
+        spacelift:
+          workspace_enabled: true
+      vars:
+        enabled: true
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/README.md:27]:
```suggestion
        enabled: true
        name: "actions-runner" # avoids hitting name length limit on IAM role
```

File: modules/eks/actions-runner-controller/default.auto.tfvars [hunk 6]
```diff
@@ -0,0 +1,24 @@
+enabled = false
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/default.auto.tfvars:1]: Delete `default.auto.tfvars` (Cloud Posse policy going forward is to remove all `auto.tfvars` files). Make sure everything is set in stack file and README.

File: modules/eks/actions-runner-controller/main.tf [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+                  path     = "/"
+                  pathType = "Prefix"
+                }
+              ]
+            }
+          ]
+        }
+      },
+      authSecret = {
+        enabled = true
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/main.tf:132]: If we are not including SOPS support now, then set 
```suggestion
        enabled = true
        create  = true
```

File: modules/eks/actions-runner-controller/resources/values.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  # with different TOKENS to get around GitHub API rate limits, among other things.
+  singleNamespace: true
+syncPeriod: 120s
+
+githubWebhookServer:
+  enabled: false
+  syncPeriod: 120s
+  secret:
+    # Webhook secret, used to authenticate incoming webhook events from GitHub
+    # Stored in same SopsSecret as authSecret under key `github_webhook_secret_token`
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/resources/values.yaml:18]:
```suggestion
    # When using Sops, stored in same SopsSecret as authSecret under key `github_webhook_secret_token`
```

File: modules/eks/actions-runner-controller/main.tf [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+    # additional values
+    yamlencode(var.chart_values)
+  ])
+
+  set_sensitive = local.enabled ? [] : [
+    {
+      name  = "authSecret.github_token"
+      value = join("", data.aws_ssm_parameter.github_token[0].*.value)
+      type  = "string"
+    }
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/main.tf:144]: Should also be setting `githubWebhookServer.secret.github_webhook_secret_token` here, when `var.webhook.enabled == true`

File: modules/eks/actions-runner-controller/outputs.tf [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+output "metadata_action_runner_releases" {
+  value       = local.enabled ? { for k, v in var.runners : k => module.actions_runner[k].metadata } : null
+  description = "Block statuses of the deployed actions-runner chart releases"
+}
+
+output "webhook_payload_url" {
+  value       = local.webhook_enabled ? format("https://${var.webhook.hostname_template}", var.tenant, var.stage, var.environment) : null
+  description = "Payload URL for GitHub webhook"
+}
```
Comment (Nuru) [inline modules/eks/actions-runner-controller/outputs.tf:14]:
```suggestion
}

```

**Review by Nuru — CHANGES_REQUESTED** [review]

File: modules/eks/actions-runner-controller/main.tf (hunk 9 above)
Comment (milldr) [reply modules/eks/actions-runner-controller/main.tf]: added, and added documentation for creating both tokens to the readme

**Review by Nuru — APPROVED** [review]

**Comment by nitrocode** [comment]
How come so many default iam actions are for the self hosted runner role ? Doesn't the role for the arc runner only need access to assume other roles ?

**Pull request merged**